inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sinks/filecollect/SenderManager.java

```diff
@@ -0,0 +1,454 @@
+        maxSenderRetry = profile.getInt(
+                CommonConstants.PROXY_SENDER_MAX_RETRY, CommonConstants.DEFAULT_PROXY_SENDER_MAX_RETRY);
+        retrySleepTime = profile.getLong(
+                CommonConstants.PROXY_RETRY_SLEEP, CommonConstants.DEFAULT_PROXY_RETRY_SLEEP);
+        isFile = profile.getBoolean(CommonConstants.PROXY_IS_FILE, CommonConstants.DEFAULT_IS_FILE);
+        offsetManager = OffsetManager.init();
+        ioThreadNum = profile.getInt(CommonConstants.PROXY_CLIENT_IO_THREAD_NUM,
+                CommonConstants.DEFAULT_PROXY_CLIENT_IO_THREAD_NUM);
+        enableBusyWait = profile.getBoolean(CommonConstants.PROXY_CLIENT_ENABLE_BUSY_WAIT,
+                CommonConstants.DEFAULT_PROXY_CLIENT_ENABLE_BUSY_WAIT);
+        batchFlushInterval = profile.getInt(PROXY_BATCH_FLUSH_INTERVAL, DEFAULT_PROXY_BATCH_FLUSH_INTERVAL);
+        authSecretId = conf.get(AGENT_MANAGER_AUTH_SECRET_ID);
+        authSecretKey = conf.get(AGENT_MANAGER_AUTH_SECRET_KEY);
+
+        this.sourcePath = sourcePath;
+        this.inlongGroupId = inlongGroupId;
+
+        this.dimensions = new HashMap<>();
+        dimensions.put(KEY_PLUGIN_ID, this.getClass().getSimpleName());
+        String metricName = String.join("-", this.getClass().getSimpleName(),
+                String.valueOf(METRIC_INDEX.incrementAndGet()));
+        this.metricItemSet = new AgentMetricItemSet(metricName);
+        MetricRegister.register(metricItemSet);
+        resendQueue = new LinkedBlockingQueue<>();
+
+    }
+
+    public void Start() throws Exception {
+        createMessageSender(inlongGroupId);
+        EXECUTOR_SERVICE.execute(flushOffset());
+        EXECUTOR_SERVICE.execute(flushResendQueue());
+        started = true;
+    }
+
+    public void Stop() {
+        LOGGER.info("stop send manager");
+        shutdown = true;
+        if (!started) {
+            return;
+        }
+        while (offsetRunning || resendRunning) {
+            AgentUtils.silenceSleepInMs(1);
+        }
+        closeMessageSender();
+        clearOffset();
+        LOGGER.info("stop send manager end");
+    }
+
+    private void closeMessageSender() {
+        if (sender != null) {
+            sender.close();
+        }
+    }
+
+    private AgentMetricItem getMetricItem(Map<String, String> otherDimensions) {
+        Map<String, String> dimensions = new HashMap<>();
+        dimensions.put(KEY_PLUGIN_ID, this.getClass().getSimpleName());
+        dimensions.putAll(otherDimensions);
+        return this.metricItemSet.findMetricItem(dimensions);
+    }
+
+    private AgentMetricItem getMetricItem(String groupId, String streamId) {
+        Map<String, String> dims = new HashMap<>();
+        dims.put(KEY_INLONG_GROUP_ID, groupId);
+        dims.put(KEY_INLONG_STREAM_ID, streamId);
+        return getMetricItem(dims);
+    }
+
+    /**
+     * sender
+     *
+     * @param tagName group id
+     * @return DefaultMessageSender
```

Comment:
no return here and what's the meaning of tagname group id ?

Reply:
the annotation is out of date